aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -623,6 +617,24 @@ void TensorIterator::mark_outputs() {
   }
 }
 
+void TensorIterator::check_overlaps() {
+  if (check_overlap_) {
+    for (int i = 0; i < num_outputs_; i++) {
+      const auto &output = operands_[i].tensor;
+      if (!output.defined()) continue;
+      assert_no_internal_overlap(output);
+    }
+    for (int arg = num_outputs_; arg < ntensors(); arg++) {
```

Comment:
This loop could be combined with the above loop

Reply:
addressed

Reply:
addressed